chef/cookbooks/barclamp/libraries/nic.rb

```diff
@@ -743,6 +744,20 @@ def slaves
       ports.map{ |i| ::Nic.new(i) }
     end
 
+    def add_slave(slave)
+      slave = self.class.coerce(slave)
+      unless ::Nic.exists?(slave)
```

Comment:
If we manage to coerce, then the slave does exist. I think it makes more sense to check for existence first, and then coerce (that's what we do in the Bond class)